src/main/java/org/spongepowered/mod/mixin/fml/MixinInjectedModContainer.java

```diff
@@ -22,19 +22,20 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+package org.spongepowered.mod.mixin.fml;
 
-package org.spongepowered.mod.guice;
+import net.minecraftforge.fml.common.InjectedModContainer;
+import net.minecraftforge.fml.common.ModContainer;
+import org.spongepowered.asm.mixin.Mixin;
 
-import com.google.inject.ScopeAnnotation;
-
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import java.lang.annotation.Target;
-
-@Target({ElementType.TYPE, ElementType.METHOD})
-@Retention(RetentionPolicy.RUNTIME)
-@ScopeAnnotation
-public @interface PluginScoped {
+/**
+ * @author zml2008
+ */
+@Mixin(InjectedModContainer.class)
+public abstract class MixinInjectedModContainer implements ModContainer {
 
+    @Override
+    public boolean isImmutable() {
+        return !getModId().equals("SpongeAPIMod");
```

Comment:
Looks like that's a possibility to do in Sponge's PluginManager. I'll do that instead.